### ISSUE TITLE
Server closed the connection unexpectedly after set maintenance_work_mem

### ISSUE THREAD

**Issue by khang-nguyen2907** [pgvector/pgvector#50]
Hi, my table contains 13 847 564 rows, I store 13 847 564  128-dimensional embedding vectors and create the index with: 
```
CREATE INDEX ON wiki_tb USING ivfflat (embedd vector_ip_ops) WITH (lists = 14885);
``` 
However, I get the error that tell me : `memory required is 43495 MB, maintenance_work_mem is 26700 MB`. Then, I set the maintenance_work_mem follows the requirement with:
```
SET maintenance_work_mem TO '43495 MB';
CREATE INDEX ON wiki_tb USING ivfflat (embedd vector_ip_ops) WITH (lists = 14885);
```
Set maintenance_work_mem works but `CREATE INDEX...` command throws the following error:
```
Error in query (7): server closed the connection unexpectedly
This probably means the server terminated abnormally 
before or while processing the request
```
I have no idea to solve this problem, could you please help me this? I would appreciate it. Thank you in advanced.

**Comment by ankane** [comment]
Hi @khang-nguyen2907, few questions to help debug:

1. What version of pgvector are you on (`SELECT * FROM pg_extension WHERE extname = 'vector';`)?
2. What installation method did you use (source, Docker, Homebrew, or PGXN)?
3. What OS are you on (Ubuntu 22.04, macOS 13, etc)?
4. How much memory does the server have?
5. What information does the Postgres server log provide about the crash?

**Comment by khang-nguyen2907** [comment]
Hi @ankane, these are answers for the questions: 
1. v0.2.7
2. Docker
3. Ubuntu 22.04
4. 125Gb
5.
```
database-pgvector-1  | 2022-12-24 06:40:12.150 GMT [1] LOG:  server process (PID 8787) was terminated by signal 11: Segmentation fault
database-pgvector-1  | 2022-12-24 06:40:12.150 GMT [1] DETAIL:  Failed process was running: CREATE INDEX ON wiki_tb USING ivfflat (embedd vector_ip_ops) WITH (lists = 14885)
database-pgvector-1  | 2022-12-24 06:40:12.150 GMT [1] LOG:  terminating any other active server processes
database-pgvector-1  | 2022-12-24 06:40:12.152 GMT [1] LOG:  all server processes terminated; reinitializing
database-pgvector-1  | 2022-12-24 06:40:12.533 GMT [8796] LOG:  database system was interrupted; last known up at 2022-12-24 06:39:27 GMT
database-pgvector-1  | 2022-12-24 06:40:12.679 GMT [8796] LOG:  database system was not properly shut down; automatic recovery in progress
database-pgvector-1  | 2022-12-24 06:40:12.681 GMT [8796] LOG:  redo starts at 14/1180C0C0
database-pgvector-1  | 2022-12-24 06:40:12.681 GMT [8796] LOG:  invalid record length at 14/1181B728: wanted 24, got 0
database-pgvector-1  | 2022-12-24 06:40:12.681 GMT [8796] LOG:  redo done at 14/1181B6F0 system usage: CPU: user: 0.00 s, system: 0.00 s, elapsed: 0.00 s
database-pgvector-1  | 2022-12-24 06:40:12.701 GMT [1] LOG:  database system is ready to accept connections```

**Comment by ankane** [comment]
That error should be fixed as of 0.3.1.
